packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,658 @@
+  readonly searchString?: string;
+}
+
+/**
+ * Amazon Route 53 health checks monitor the health and performance of your web applications, web servers, and other resources. Each health check that you create can monitor one of the following:
+ * - The health of a resource, such as a web server,
+ * - The status of other health checks, and
+ * - The CloudWatch alarm that you specify,
+ * - The status of an Amazon Route 53 routing control.
+ */
+export class HealthCheck extends Resource implements IHealthCheck {
+  /**
+   * Import an existing health check into this CDK app.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param healthCheckId ID of the health check.
+   * @returns a reference to the existing health check.
+   */
+  public static fromHealthCheckId(scope: Construct, id: string, healthCheckId: string): IHealthCheck {
+    class Import extends Resource implements IHealthCheck {
+      public readonly healthCheckId = healthCheckId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly healthCheckId: string;
+
+  /**
+   * Creates a new health check.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param props the properties of the new health check.
+   * @returns a reference to the newly created health check.
+   */
+  constructor(scope: Construct, id: string, props: HealthCheckProps) {
+    super(scope, id);
+
+    validateProperties(props);
+
+    const resource = new CfnHealthCheck(this, 'Resource', {
+      healthCheckConfig: {
+        type: props.type,
+        alarmIdentifier: props.alarmIdentifier,
+        childHealthChecks: props.childHealthChecks?.map((childHealthCheck) => childHealthCheck.healthCheckId),
+        enableSni: props.enableSNI ?? getDefaultEnableSNIForType(props.type),
+        failureThreshold: props.failureThreshold ?? getDefaultFailureThresholdForType(props.type),
+        fullyQualifiedDomainName: props.fqdn,
+        healthThreshold: props.healthThreshold ?? getDefaultHealthThresholdForType(props),
+        insufficientDataHealthStatus: props.insufficientDataHealthStatus ?? getDefaultInsufficientDataHealthStatusForType(props.type),
+        inverted: props.inverted ?? false,
+        ipAddress: props.ipAddress,
+        measureLatency: props.measureLatency ?? getDefaultMeasureLatencyForType(props.type),
+        port: props.port ?? getDefaultPortForType(props.type),
+        regions: props.regions,
+        requestInterval: props.requestInterval?.toSeconds() ?? getDefaultRequestIntervalForType(props.type)?.toSeconds(),
+        resourcePath: props.resourcePath ?? getDefaultResourcePathForType(props.type),
+        routingControlArn: props.routingControl,
+        searchString: props.searchString,
+      },
+    });
+
+    this.healthCheckId = resource.ref;
+  }
+}
+
+const validationRules: Record<HealthCheckType, ((props: HealthCheckProps) => void)[]> = {
```

Comment:
instead of having all these functions to validate these scenarios, can we include something like a enum based properties or class to define these props, i think one of the [design principles](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#property-documentation) we follow is to prevent many check cases.

The fewer “if statements” the better.

Reply:
sorry @shikha372 could you please explain this further?

Reply:
referring to all the validation functions defined with each healthcheck type eg. 
```
function ruleAlarmIdentifierIsNotAllowed(props: HealthCheckProps) {
  if (props.alarmIdentifier) {
    throw new Error(`AlarmIdentifier is not supported for health check type: ${props.type}`);
  }
}
```
instead of validating these checks with if-else, can we introduce a enum type class or method which will allow users to set only the required fields possible under a healthcheck, let me know if you think this is not feasible